src/test/java/seedu/address/logic/parser/meeting/EditMeetingCommandParserTest.java

```diff
@@ -76,28 +87,43 @@ public void parse_invalidPreamble_failure() {
 
     @Test
     public void parse_invalidValue_failure() {
+        // invalid name
+        assertParseFailure(parser, "1" + INVALID_NAME_DESC, MeetingName.MESSAGE_CONSTRAINTS);
+
+        // invalid url
+        assertParseFailure(parser, "1" + INVALID_URL_DESC, MeetingUrl.MESSAGE_CONSTRAINTS);
+
+        // invalid datetime
+        assertParseFailure(parser, "1" + INVALID_DATETIME_DESC, MeetingDateTime.MESSAGE_CONSTRAINTS);
+
         // invalid duration
         assertParseFailure(parser, "1" + INVALID_DURATION_DESC, MeetingDuration.MESSAGE_CONSTRAINTS);
 
-        // TODO: Add more tests once we have more constraints on what is valid.
+        // invalid module
+        assertParseFailure(parser, "1" + INVALID_MODULE_DESC, ParserUtil.MESSAGE_INVALID_INDEX);
+
+        // invalid isRecurring
+        assertParseFailure(parser, "1" + INVALID_RECURRING_DESC, IsRecurring.MESSAGE_CONSTRAINTS);
+
+        // invalid tag
+        assertParseFailure(parser, "1" + INVALID_TAG_DESC, Tag.MESSAGE_CONSTRAINTS);
     }
 
-    // TODO MODULE INDEX: fix
-    //@Test
-    //public void parse_allFieldsSpecified_success() {
-    //    final Index targetIndex = INDEX_SECOND_MEETING;
-    //    final String userInput = targetIndex.getOneBased() + NAME_DESC_LECTURE + URL_DESC_TUTORIAL
-    //            + DATETIME_DESC_LECTURE + DURATION_DESC_LECTURE + MODULE_DESC_LECTURE + RECURRING_DESC_LECTURE
-    //            + TAG_DESC_LECTURE;
-    //
-    //    final EditMeetingDescriptor descriptor = new EditMeetingDescriptorBuilder().withName(VALID_NAME_LECTURE)
-    //            .withUrl(VALID_URL_TUTORIAL).withDateTime(VALID_DATETIME_LECTURE).withDuration(VALID_DURATION_LECTURE)
-    //            .withModule(VALID_MODULE_LECTURE).withIsRecurring(VALID_RECURRING_LECTURE)
-    //            .withTags(VALID_TAG_LECTURE).build();
-    //    final EditMeetingCommand expectedCommand = new EditMeetingCommand(targetIndex, descriptor);
-    //
-    //    assertParseSuccess(parser, userInput, expectedCommand);
-    //}
+    @Test
+    public void parse_allFieldsSpecified_success() {
+        final Index targetIndex = INDEX_SECOND_MEETING;
+        final String userInput = targetIndex.getOneBased() + NAME_DESC_LECTURE + URL_DESC_TUTORIAL
+                + DATETIME_DESC_LECTURE + DURATION_DESC_LECTURE + MODULE_DESC_LECTURE + RECURRING_DESC_LECTURE
+                + TAG_DESC_LECTURE;
+
+        final EditMeetingDescriptor descriptor = new EditMeetingDescriptorBuilder().withName(VALID_NAME_LECTURE)
+                .withUrl(VALID_URL_TUTORIAL).withDateTime(VALID_DATETIME_LECTURE).withDuration(VALID_DURATION_LECTURE)
+                .withModule(Index.fromZeroBased(1)).withIsRecurring(VALID_RECURRING_LECTURE)
```

Comment:
```suggestion
                .withModule(VALID_MODULE_INDEX_LECTURE).withIsRecurring(VALID_RECURRING_LECTURE)
```

If the earlier suggestions are taken up this can be imported and used here